.github/workflows/comment-on-pr-with-build-url.yml

```diff
@@ -0,0 +1,29 @@
+name: Comment On PR With Build URL
+on:
+  workflow_run:
+    workflows: ['Slash Command Build']
+    types: [completed]
+jobs:
+  commentOnPrWithBuildUrl:
+    if: github.event.workflow_run.event == 'repository_dispatch' && github.event.workflow_run.conclusion == 'success'
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Fetch artifact
+        id: fetch_artifact
+        run: |
+          artifact=$(curl -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/${{github.repository}}/actions/runs/${{github.event.workflow_run.id}}/artifacts | jq .artifacts[0])
+          artifact_id=$(jq -r '.id' <<< "${artifact}")
+          artifact_name=$(jq -r '.name' <<< "${artifact}")
+          issue_id=${artifact_name##*-}
+          echo "::set-output name=artifact_id::${artifact_id}"
+          echo "::set-output name=issue_id::${issue_id}"
+
+      - name: Create comment
+        uses: peter-evans/create-or-update-comment@v1
+        with:
+          repository: ${{github.repository}}
+          issue-number: ${{steps.fetch_artifact.outputs.issue_id}}
+          body: |
+            Your build is ready.
+
+            [woocommerce-payments.zip](https://nightly.link/${{github.repository}}/actions/artifacts/${{steps.fetch_artifact.outputs.artifact_id}}.zip)
```

Comment:
Note the use of the `nightly.link` link here. This is a free service that makes it a lot easier to work with GitHub artifacts.

You can read more here → https://nightly.link/

> Because GitHub doesn't provide any permanent and public links to an artifact, this service redirects to time-limited links that GitHub can give to the application -- only on behalf of an authenticated user that has access to the repository. So, whenever someone downloads an artifact from a repository that you had added, this service uses a token that is associated with your installation of the GitHub App.

Reply:
Sounds like this is worth documenting / justifying in a comment!

I took a quick look at the docs and I don't fully understand why this is needed – i.e. what the gap is in GitHub's features. Is it that the url doesn't work without being logged in to GitHub? If so, maybe we can solve that with a deploy key for the consuming system (e.g. test environment).

If this is needed I have no objections, just exploring the rationale so I fully understand.